autoload/lsp/internal/inlay_hints.vim

```diff
@@ -0,0 +1,95 @@
+let s:use_vim_textprops = lsp#utils#_has_textprops() && !has('nvim')
+
+function! s:set_inlay_hints(data) abort
+    let l:bufnr = bufnr('%')
+
+    call s:clear_inlay_hints()
+
+    if mode() !=# 'n' | return | endif
+
+    if lsp#client#is_error(a:data['response']) | return | endif
+
+    " Get hints from the response
+    let l:hints = a:data['response']['result']
+    if empty(l:hints)
+        return
+    endif
+
+    for l:hint in l:hints
+        let l:text = (get(l:hint, 'paddingLeft', v:false) ? ' ' : '') . l:hint.label[0].value . (get(l:hint, 'paddingRight', v:false) ? ' ' : '')
+        if l:hint.kind ==# 1
+            call prop_add(l:hint.position.line+1, l:hint.position.character+1, {'type': 'vim_lsp_inlay_hint_type', 'text': l:text, 'bufnr': l:bufnr})
+        elseif l:hint.kind ==# 2
+            call prop_add(l:hint.position.line+1, l:hint.position.character+1, {'type': 'vim_lsp_inlay_hint_parameter', 'text': l:text, 'bufnr': l:bufnr})
+        endif
+    endfor
+endfunction
+
+function! s:init_inlay_hints() abort
+    if index(prop_type_list(), 'vim_lsp_inlay_hint') == -1
+        call prop_type_add('vim_lsp_inlay_hint_type', { 'highlight': 'Label' })
+        call prop_type_add('vim_lsp_inlay_hint_parameter', { 'highlight': 'Todo' })
+    endif
+endfunction
+
+function! lsp#internal#inlay_hints#_disable() abort
+    if exists('s:Dispose')
+        call s:Dispose()
+        unlet s:Dispose
+    endif
+endfunction
+
+function! s:clear_inlay_hints() abort
+    let l:bufnr = bufnr('%')
+    for l:prop in prop_list(1, {'end_lnum': line('$'), 'types': ['vim_lsp_inlay_hint_type'], 'bufnr': l:bufnr})
```

Comment:
should we call `prop_remove({ 'type': 'vim_lsp_inlay_hint_type' })` instead? Would this be faster instead of looping  in the vimscript?